5.1RC1: test_models_fitting[LevMarLSQFitter-…] fail on several architectures

### Description

Testing Debian build for 5.1RC1 

When running tests (during build time) on i386 (32 bit), the tests test_models_fitting[LevMarLSQFitter-model5] and test_models_fitting[LevMarLSQFitter-model5] fail.

### Actual behavior

```
_________________ test_models_fitting[LevMarLSQFitter-model5] __________________
model = {'bounding_box': <Quantity [-3.,  3.] km / s>, 'class': <class 'astropy.modeling.functional_models.ArcSine1D'>
Name: A..., <Quantity -2. s>)], 'parameters': {'amplitude': <Quantity 3. km / s>, 'frequency': <Quantity 0.25 Hz>, 'phase': 0.5}}
fitter = <astropy.modeling.fitting.LevMarLSQFitter object at 0xe8631568>
[…]
    def test_models_fitting(model, fitter):
        fitter = fitter()
[…]
        m = model['class'](**model['parameters'])
        if len(model['evaluation'][0]) == 2:
            x = np.linspace(1, 3, 100) * model['evaluation'][0][0].unit
            y = np.exp(-x.value ** 2) * model['evaluation'][0][1].unit
            args = [x, y]
        else:
            x = np.linspace(1, 3, 100) * model['evaluation'][0][0].unit
            y = np.linspace(1, 3, 100) * model['evaluation'][0][1].unit
            z = np.exp(-x.value**2 - y.value**2) * model['evaluation'][0][2].unit
            args = [x, y, z]
    
        # Test that the model fits even if it has units on parameters
>       m_new = fitter(m, *args)
astropy/modeling/tests/test_models_quantities.py:566: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
astropy/modeling/fitting.py:253: in wrapper
    model_new = func(self, model, xdata, ydata, **kwargs)
astropy/modeling/fitting.py:1222: in __call__
    init_values, fitparams, cov_x = self._run_fitter(model_copy, farg,
astropy/modeling/fitting.py:1283: in _run_fitter
    fitparams, cov_x, dinfo, mess, ierr = optimize.leastsq(
/usr/lib/python3/dist-packages/scipy/optimize/minpack.py:432: in leastsq
    retval = _minpack._lmder(func, Dfun, x0, args, full_output,
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = <astropy.modeling.fitting.LevMarLSQFitter object at 0xe8631568>
fps = array([nan, nan, nan])
args = (<ArcSine1D(amplitude=nan, frequency=nan, phase=nan)>, None, array([1.        , 1.02020202, 1.04040404, 1.06060606, 1.....51663004e-04, 2.23937753e-04, 1.99104358e-04,
       1.76880405e-04, 1.57008872e-04, 1.39256082e-04, 1.23409804e-04]))
model = <ArcSine1D(amplitude=nan, frequency=nan, phase=nan)>, weights = None
meas = array([3.67879441e-01, 3.53167736e-01, 3.38767730e-01, 3.24689734e-01,
       3.10942862e-01, 2.97535051e-01, 2.844730...2.51663004e-04, 2.23937753e-04, 1.99104358e-04,
       1.76880405e-04, 1.57008872e-04, 1.39256082e-04, 1.23409804e-04])
    def objective_function(self, fps, *args):
        """…"""
    
        model = args[0]
        weights = args[1]
        fitter_to_model_params(model, fps, self._use_min_max_bounds)
        meas = args[-1]
    
        if weights is None:
            value = np.ravel(model(*args[2: -1]) - meas)
        else:
            value = np.ravel(weights * (model(*args[2: -1]) - meas))
    
        if not np.all(np.isfinite(value)):
>           raise NonFiniteValueError("Objective function has encountered a non-finite value, "
                                      "this will cause the fit to fail!")
E           astropy.modeling.fitting.NonFiniteValueError: Objective function has encountered a non-finite value, this will cause the fit to fail!
astropy/modeling/fitting.py:1077: NonFiniteValueError
```
Full build log [here](https://buildd.debian.org/status/fetch.php?pkg=astropy&arch=i386&ver=5.1%7Erc1-1&stamp=1651578947&raw=0).
This happens for Python 3.9 and Python 3.10. It does not appear on x86_64.

### Steps to Reproduce

Build the 5.1RC1 version and run the tests on a 32-bit x86.

### System Details

Happens on [i386](https://buildd.debian.org/status/fetch.php?pkg=astropy&arch=i386&ver=5.1%7Erc1-1&stamp=1651578947&raw=0), [s390 (64 bit big endian)](https://buildd.debian.org/status/fetch.php?pkg=astropy&arch=s390x&ver=5.1%7Erc1-1&stamp=1651578129&raw=0), [ppc64el (64 bit little endian)](https://buildd.debian.org/status/fetch.php?pkg=astropy&arch=ppc64el&ver=5.1%7Erc1-1&stamp=1651579476&raw=0) (fere also LMLSQFitter), [armhf (ARM 32 bit little endian](https://buildd.debian.org/status/fetch.php?pkg=astropy&arch=armhf&ver=5.1%7Erc1-1&stamp=1651580401&raw=0), [arm64 (64 bit little endian)](https://buildd.debian.org/status/fetch.php?pkg=astropy&arch=arm64&ver=5.1%7Erc1-1&stamp=1651580387&raw=0) (also LMLSQFitter),
Not affected: x86_64, MIPS 32 bit little endian, 

Platform linux -- Python 3.10.4, pytest-6.2.5, py-1.10.0, pluggy-1.0.0
Running tests with Astropy version 5.1-rc1.
Running tests in .pybuild/cpython3_3.10/build.
Date: 2022-05-03T10:16:27
**Platform: Linux-5.10.0-13-cloud-amd64-x86_64-with-glibc2.33**
Executable: /usr/bin/python3.10
Full Python Version: 
**3.10.4 (main, Apr  2 2022, 09:04:19) [GCC 11.2.0]**
encodings: sys: utf-8, locale: UTF-8, filesystem: utf-8
byteorder: little
float info: dig: 15, mant_dig: 15
Package versions: 
**Numpy: 1.21.5**
**Scipy: 1.7.3**
**Matplotlib: 3.5.1**
h5py: 3.6.0
Pandas: not available
**PyERFA: 2.0.0.1**
Cython: 0.29.28
Scikit-image: 0.18.3
asdf: not available
pyarrow: not available
Using Astropy options: remote_data: none.
Matplotlib: 3.5.1
Freetype: 2.11.1
ARCH_ON_CI: undefined
IS_CRON: undefined
rootdir: /builds/debian-astro-team/astropy/debian/output/source_dir, configfile: setup.cfg
plugins: arraydiff-0.5.0, astropy-header-0.2.1, filter-subpackage-0.1.1, doctestplus-0.12.0, remotedata-0.3.3, openfiles-0.5.0, mpl-0.11, mock-3.6.1, hypothesis-6.36.0, cov-3.0.0, astropy-0.10.0

--- a/astropy/modeling/tests/test_models_quantities.py
+++ b/astropy/modeling/tests/test_models_quantities.py
@@ -350,8 +350,11 @@ SCIPY_MODELS = set([Sersic1D, Sersic2D, AiryDisk2D])
 #   will produce non-finite values
 NON_FINITE_LevMar_MODELS = [
     Sersic1D,
+    ArcSine1D,
+    ArcCosine1D,
     PowerLaw1D,
     ExponentialCutoffPowerLaw1D,
+    BrokenPowerLaw1D,
     LogParabola1D
 ]
 
@@ -372,7 +375,9 @@ NON_FINITE_LM_MODELS = [
     ArcSine1D,
     ArcCosine1D,
     PowerLaw1D,
-    LogParabola1D
+    LogParabola1D,
+    ExponentialCutoffPowerLaw1D,
+    BrokenPowerLaw1D
 ]
 
 # These models will fail the DogBoxLSQFitter fitting test due to non-finite

